tools/fixedpointarithmetic/src/test/java/dk/alexandra/fresco/fixedpoint/basic/BasicFixedPointTests.java

```diff
@@ -0,0 +1,440 @@
+package dk.alexandra.fresco.fixedpoint.basic;
+
+import dk.alexandra.fresco.decimal.RealNumeric;
+import dk.alexandra.fresco.decimal.SReal;
+import dk.alexandra.fresco.decimal.fixed.FixedNumeric;
+import dk.alexandra.fresco.framework.Application;
+import dk.alexandra.fresco.framework.DRes;
+import dk.alexandra.fresco.framework.TestThreadRunner.TestThread;
+import dk.alexandra.fresco.framework.TestThreadRunner.TestThreadFactory;
+import dk.alexandra.fresco.framework.builder.numeric.ProtocolBuilderNumeric;
+import dk.alexandra.fresco.framework.sce.resources.ResourcePool;
+import dk.alexandra.fresco.framework.value.SInt;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.Assert;
+
+public class BasicFixedPointTests {
+
+  public static class TestInput<ResourcePoolT extends ResourcePool>
+      extends TestThreadFactory<ResourcePoolT, ProtocolBuilderNumeric> {
+
+    @Override
+    public TestThread<ResourcePoolT, ProtocolBuilderNumeric> next() {
+      BigDecimal value = BigDecimal.valueOf(10.00100);
```

Comment:
Please use _ in long numbers
https://docs.oracle.com/javase/8/docs/technotes/guides/language/underscores-literals.html